config/oidc.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import "github.com/ngrok/ngrok-go/internal/pb_agent"
+
+type OIDCOption func(cfg *oidcOptions)
+
+type oidcOptions struct {
+	IssuerURL    string
+	ClientID     string
+	ClientSecret string
+	AllowEmails  []string
+	AllowDomains []string
+	Scopes       []string
+}
+
+func (oidc *oidcOptions) toProtoConfig() *pb_agent.MiddlewareConfiguration_OIDC {
+	if oidc == nil {
+		return nil
+	}
+
+	return &pb_agent.MiddlewareConfiguration_OIDC{
+		IssuerURL:    oidc.IssuerURL,
+		ClientID:     oidc.ClientID,
+		ClientSecret: oidc.ClientSecret,
+		AllowEmails:  oidc.AllowEmails,
+		AllowDomains: oidc.AllowDomains,
+		Scopes:       oidc.Scopes,
+	}
+}
+
+// WithOIDC configures this edge with the the given OIDC provider.
+// Overwrites any previously-set OIDC configuration.
+func WithOIDC(issuerURL string, clientID string, clientSecret string, opts ...OIDCOption) HTTPEndpointOption {
+	return httpOptionFunc(func(cfg *httpOptions) {
+		oidc := &oidcOptions{
+			IssuerURL:    issuerURL,
+			ClientID:     clientID,
+			ClientSecret: clientSecret,
+		}
+
+		for _, opt := range opts {
+			opt(oidc)
+		}
+
+		cfg.OIDC = oidc
+	})
+}
+
+// Append email addresses to the list of allowed emails.
+func WithOIDCEmail(addr ...string) OIDCOption {
```

Comment:
should this (and the ones for oauth too) be `WithOIDCAllowEmail`, `WithOAuthAllowDomain`, etc? i think so